pipeline/frontend/yaml/compiler/convert.go

```diff
@@ -114,8 +113,8 @@ func (c *Compiler) createProcess(container *yaml_types.Container, stepType backe
 		return secret.Value, nil
 	}
 
-	// TODO: why don't we pass secrets to detached steps?
-	if !detached {
+	// TODO: why don't we pass settings to services?
+	if stepType != backend_types.StepTypeService {
```

Comment:
good question ... I would pass them down if they exist ... but we can address that in a new pull that also alters the json schema and docs